### ISSUE TITLE
Proxy silently fails when trying to open an instagram profile because of a cookie host mismatch error

### ISSUE THREAD

**Issue by ardalann** [Cuadrix/puppeteer-page-proxy#32]
**Steps to reproduce:**
1. Use this package to proxy a **request** to https://instagram.com/nike 

**Expected result:** The profile loads, or a login form shows up if the proxy server is not trusted by Instagram

**Actual result:** The proxy package never resolves the request. It just fails silently. 

**Solution:** Turns our it was a cookie host mismatch error that was being caught by the package. I fixed this issue by adding `ignoreInvalidCookies: true` to `got`'s `options` in [proxy.js](https://github.com/Cuadrix/puppeteer-page-proxy/blob/47358301a1366389315dcfc9cfc34b45430d3bf6/src/core/proxy.js#L22).

---
But now I have a different issue: Instagram keeps redirecting me to the same URL. The redirect is client-side, so it happens infinitely. 

This issue does NOT happen when I use the proxy-server arg with Puppeteer, so it must have something to do with this package.

I suspect it has something to do with how this package handle cookies.

I'll update this issue as I investigate this further. Just wanted to post here so others don't waste hours getting this far like I did.

**Comment by jackwicklow1** [comment]
Hi I am a beginner and I am struggling to find the right structure in order to get the code of puppeteer-page-proxy working. Is there any chance you could show the code you used to use proxies?
Thanks

**Comment by ardalann** [comment]
Turns out the infinite client-side redirects was caused by this:
1. Instagram redirects to its login page
2. `got` handles the redirect and sends the response body and headers back to Puppeteer, **but not the final page url**
3. Instagram's SPA Javascript tries to run assuming the URL is the original URL, not knowing the user has been redirected

The issue is that `got` is handling server-side redirects. Puppeteer should be handling them.

I fixed this issue and created a PR: https://github.com/Cuadrix/puppeteer-page-proxy/pull/35
